edsby.py

```diff
@@ -515,39 +615,52 @@ def getClassAssignmentList(self, classNID, classRID):
             assignmentNID = str(metadata[assg]['nid'])
 
             # Copy other keys from metadata obj to compiled assignment data obj
-            for key in metadata['r'+str(metadata[assg]['rid'])]:
-                if assignmentNID in assignmentData['assignments']: # If we've found metadata for an asssignment that we've also found scores for
-                    assignmentData['assignments'][assignmentNID][key] = metadata[assg][key] # Copy metadata to that entry
+            for key in metadata['r' + str(metadata[assg]['rid'])]:
+                if assignmentNID in assignmentData[
+                    'assignments']:  # If we've found metadata for an asssignment that we've also found scores for
+                    assignmentData['assignments'][assignmentNID][key] = metadata[assg][
+                        key]  # Copy metadata to that entry
                 else:
-                    assignmentData['no_scores_found'][assignmentNID] = dict() # Otherwise, place this metadata in the no_scores_found dict
+                    assignmentData['no_scores_found'][
+                        assignmentNID] = dict()  # Otherwise, place this metadata in the no_scores_found dict
                     for meta in metadata[assg]:
                         assignmentData['no_scores_found'][assignmentNID][meta] = metadata[assg][meta]
 
         # Copy weighting data, sort assignments without it, calculate percentage scores if possible
         for assg in assignmentData['assignments']:
             assignmentNID = str(assg)
 
-            if 'weighting' in assignmentData['assignments'][assg]: # If weighting data is present in the metadata we retrieved
+            if 'weighting' in assignmentData['assignments'][
+                assg]:  # If weighting data is present in the metadata we retrieved
                 # API sometimes returns a dict, other times returns a JSON string. Figure out which one it is and parse appropriately.
-                if isinstance(assignmentData['assignments'][assg]['weighting'], dict): # If dict access weighting prop as a dict
-                    assignmentData['assignments'][assignmentNID]['weighting'] = assignmentData['assignments'][assg]['weighting']['0']
-
-                elif isinstance(assignmentData['assignments'][assg]['weighting'], basestring): # If string access weighting prop as a dict after running through a JSON parser
-                    assignmentData['assignments'][assignmentNID]['weighting'] = json.loads(assignmentData['assignments'][assg]['weighting'])['0']
+                if isinstance(assignmentData['assignments'][assg]['weighting'],
+                              dict):  # If dict access weighting prop as a dict
+                    assignmentData['assignments'][assignmentNID]['weighting'] = \
```

Comment:
For long variable assignments, I could appreciate having them split into mutliple lines, so this is a formatting change that I could consider useful.